analysisservice/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -132,53 +158,61 @@ private void analyze(ActivityPayload payload, Set<ActivityCategoryDTO> matchingA
 		// because the lock is per user, it doesn't matter much that we block early.
 		try (LockPool<UUID>.Lock lock = userAnonymizedSynchronizer.lock(payload.userAnonymized.getID()))
 		{
-			analyzeInsideLock(payload, matchingActivityCategories);
+			UserAnonymizedEntityHolder userAnonymizedHolder = new UserAnonymizedEntityHolder(payload.userAnonymized.getID());
+			analyzeInsideLock(userAnonymizedHolder, payload, matchingActivityCategories);
+			if (userAnonymizedHolder.isEntityFetched())
+			{
+				userAnonymizedService.updateUserAnonymized(userAnonymizedHolder.getEntity());
+			}
 		}
 	}
 
-	private void analyzeInsideLock(ActivityPayload payload, Set<ActivityCategoryDTO> matchingActivityCategories)
+	private void analyzeInsideLock(UserAnonymizedEntityHolder userAnonymizedHolder, ActivityPayload payload,
```

Comment:
This results in 1 extra `userAnonymizedHolder` argument per method. Can't the `userAnonymizedHolder` be included in the `ActivityPayload`?